src/NuGet.Clients/NuGet.PackageManagement.VisualStudio/IDE/VSSolutionManager.cs

```diff
@@ -205,6 +206,24 @@ private async Task InitializeAsync()
             var solutionLoadedGuid = VSConstants.UICONTEXT.SolutionExistsAndFullyLoaded_guid;
             _vsMonitorSelection.GetCmdUIContextCookie(ref solutionLoadedGuid, out _solutionLoadedUICookie);
 
+            if (ErrorHandler.Succeeded(_vsSolution.GetProperty((int)__VSPROPID.VSPROPID_IsSolutionOpen, out object isSoltuionOpen)))
```

Comment:
Let's say if the call to `_vsSolution.GetProperty` fails due to unknown reason. For example, see [internal issue](https://devdiv.visualstudio.com/DevDiv/_workitems/edit/1593769). Is there any chance that `_solutionEventsCookie != 0`? If yes, then `IsSolutionOpenAsync()` or `SolutionDirectory` always returns the default values. I think in the property getters or methods that relay on `_solutionEventsCookie` should also check for both cases. For example, `SolutionDirectory` getter checks for ` if (_solutionEventsCookie != 0 && _solutionDirectory !=null)` before returning the solution directory from the cache. Please correct if I have misunderstood something.
```cs
        public string SolutionDirectory
        {
            get
            {
                if (_solutionEventsCookie != 0 && _solutionDirectory !=null)
                {
                    return _solutionDirectory;
                }
                else
                {
                    return NuGetUIThreadHelper.JoinableTaskFactory.Run(GetSolutionDirectoryAsync);
                }
            }
        }
```

Reply:
While there aren't many scenarios where NuGet will try to get the solution directory when there is no open solution, I believe we should still avoid switching to the UI thread, and stop using JTF.Run on a background thread.

My understanding of .NET COM interop is that calls can't throw exceptions. They just return an `int` value that signals an error, which is exactly what `ErrorHandler.Success(value)` checks, and which I'm intentionally not using `ErrorHandler.ThrowOnFailure(value)`, since getting the `VSPROPID_SolutionDirectory` returns a non-successful error code when the solution is closed, or when there's a temporary solution with no directory.